service/controller/resource/monitoring/scrapeconfigs/resource.go

```diff
@@ -195,13 +199,47 @@ func getTemplateData(cluster metav1.Object, config Config) (*TemplateData, error
 	return d, nil
 }
 
-// HasPrometheusAgent returns true if the release uses the prometheus agent to collect k8s metrics. The prometheus agent will be added in v19, so any release >= v19.0.0
-func HasPrometheusAgent(cluster metav1.Object, provider string, installation string) (bool, error) {
-	if key.IsCAPIManagementCluster(provider) || key.IsInCluster(installation, cluster) {
-		// Bundle is currently not deployed in CAPI and for Vintage CAPI MC
+func getDefaultAppVersion(ctx context.Context, ctrlClient client.Client, cluster metav1.Object, config Config) (string, error) {
+	appName := fmt.Sprintf("%s-default-apps", key.ClusterID(cluster))
+	appNamespace := fmt.Sprintf("org-%s", key.GetOrganization(cluster))
+	objectKey := types.NamespacedName{Namespace: appNamespace, Name: appName}
+
+	app := &appsv1alpha1.App{}
+	err := ctrlClient.Get(ctx, objectKey, app)
+	if err != nil {
+		return "", err
+	}
+	return app.Status.Version, nil
+}
+
+// hasPrometheusAgent returns true if the release uses the prometheus agent to collect k8s metrics.
+func hasPrometheusAgent(ctx context.Context, ctrlClient client.Client, cluster metav1.Object, config Config) (bool, error) {
+	// For CAPI clusters, this is a case to case basis. We need to check the default app version for now.
+	if key.IsCAPIManagementCluster(config.Provider) {
+		appVersion, err := getDefaultAppVersion(ctx, ctrlClient, cluster, config)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+		version, err := semver.Parse(appVersion)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+		switch config.Provider {
+		case "openstack":
+			openstackAgentVersion, err := semver.Parse("0.8.0")
+			if err != nil {
+				return false, microerror.Mask(err)
+			}
+			return version.GTE(openstackAgentVersion), nil
```

Comment:
If openstack 0.8.0 + we can assume it's running the agent

Reply:
Done